packages/tfchain_client/lib/src/client.dart

```diff
@@ -0,0 +1,183 @@
+part of '../tfchain_client.dart';
+
+class QueryClient {
+  final String url;
+  late final Provider? provider;
+  late final polkadot.Dev api;
+  late final QueryContracts contracts;
+  late final balance.QueryBalances balances;
+  late final QueryFarms farms;
+  late final QueryNodes nodes;
+  late final QueryPricingPolicies policies;
+  late final QueryTwins twins;
+  late final QueryBridge bridge;
+  late final Dao.QueryDao dao;
+  late final QueryTFTPrice price;
+
+  QueryClient(this.url) {
+    provider = Provider.fromUri(Uri.parse(url));
+    api = polkadot.Dev(provider!);
+    contracts = QueryContracts(this);
+    balances = balance.QueryBalances(this);
+    farms = QueryFarms(this);
+    nodes = QueryNodes(this);
+    policies = QueryPricingPolicies(this);
+    twins = QueryTwins(this);
+    bridge = QueryBridge(this);
+    price = QueryTFTPrice(this);
+    dao = Dao.QueryDao(this);
+  }
+
+  void checkInputs() {
+    if (url.isEmpty) {
+      throw FormatException("URL should be provided");
+    }
+  }
+
+  Future<void> connect() async {
+    checkInputs();
+  }
```

Comment:
connect method doesn't look like it's creating a connection

Reply:
would you prefer adding this line ` provider = Provider.fromUri(Uri.parse(url));
` in connect method rather than constructor ?